firmware/library/L2_Utilities/utility_test.cpp

```diff
@@ -0,0 +1,104 @@
+#include "L2_Utilities/utility.hpp"
+#include "L5_Testing/testing_frameworks.hpp"
+
+TEST_CASE("Testing Random Functions", "[random]")
+{
+    SECTION("SetRandomSeed Consistency")
+    {
+        utility::SetRandomSeed();
+        uint32_t randomtester1 = std::rand() % 100;
+        utility::SetRandomSeed();
+        uint32_t randomtester2 = std::rand() % 100;
+        CHECK(randomtester1 == randomtester2);
+    }
+
+    SECTION("Random")
+    {
+        utility::SetRandomSeed();
+        uint32_t randomtester1 = std::rand() % 100;
+        utility::SetRandomSeed();
+        uint32_t randomtester2 = utility::Random(0, 99);
+        CHECK(randomtester1 == randomtester2);
```

Comment:
Should check that the next few random numbers do not equal the first number.

Reply:
Should also check that random numbers fall within the range, Run a for loop for 1000 numbers within a really narrow size, lets say 4 bits, or between 1 and 15. Make sure that the number never above 15 or below 1.